### PR TITLE
Kafka topic release 3.9.0

### PR DESCRIPTION
**Pull request by sivakarthikeyanoptit**
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

### Type of change

Please choose appropriate options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes in the below checkboxes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Ran Test A
- [ ] Ran Test B

**Test Configuration**:
* Software versions:
* Hardware versions:

### Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by G33tha — COMMENTED** [review]
remove the hardcoded dev value for kafka topics name, and dont change the existing variable env. also keep the tag as it is for all the tasks and update when condition (if ml-service-deploy is set to true)

File: ansible/roles/setup-kafka/tasks/main.yml [hunk 1]
```diff
@@ -12,3 +12,11 @@
...
   tags:
     - processing-kafka
+
+- name: create topics
+  command: /opt/kafka/bin/kafka-topics.sh --zookeeper localhost:2181 --create --topic {{env_name}}.{{item.name}} --partitions {{ item.num_of_partitions }} --replication-factor {{ item.replication_factor }}
+  with_items: "{{ml_service_topics}}"
+  ignore_errors: true
+  when: kafka_id=="1"
+  tags:
+    - ml-service-topics
```
Comment (G33tha) [inline ansible/roles/setup-kafka/tasks/main.yml:22]: keep the tag name as it is)processing-kafka)  and update the when condition (check if ml-servive-deploy is set to true)

File: ansible/roles/setup-kafka/defaults/main.yml [hunk 2]
```diff
@@ -27,4 +28,50 @@ processing_kafka_topics:
   - name: lms.notification
     num_of_partitions: 1
     replication_factor: 1
+ 
 
+ml_service_topics:
+  - name: dev.sl.observation.raw
```
[the remark this answers is not in the document]
Comment (sivakarthikeyanoptit) [reply ansible/roles/setup-kafka/defaults/main.yml]: done

File: ansible/roles/setup-kafka/tasks/main.yml (hunk 1 above)
Comment (sivakarthikeyanoptit) [reply ansible/roles/setup-kafka/tasks/main.yml]: changed the tag name

Comment (sivakarthikeyanoptit) [reply ansible/roles/setup-kafka/tasks/main.yml]: when condtion unable to keep like this
when: kafka_id=="1" and ml_service_topics.deploy == true
 when: kafka_id=="1" and ml_service_topics is defined
so i have defined like this is ok @G33tha

File: ansible/roles/setup-kafka/tasks/main.yml [hunk 3]
```diff
@@ -12,3 +12,11 @@
   when: kafka_id=="1" and item.retention_time is defined  
   tags:
     - processing-kafka
+
+- name: create topics
```
Comment (keshavprasadms) [inline ansible/roles/setup-kafka/tasks/main.yml:16]: Why do we need another block when you can just list in the previous task itself?
```
with_items: 
  - "{{processing_kafka_topics}}"
  - "{{ml_service_topics}}"
```

File: ansible/roles/setup-kafka/defaults/main.yml [hunk 4]
```diff
@@ -27,4 +30,50 @@ processing_kafka_topics:
   - name: lms.notification
     num_of_partitions: 1
     replication_factor: 1
+ 
 
+ml_service_topics:
+  - name: sl.observation.raw
+    retention_time: 604800000
```
Comment (keshavprasadms) [inline ansible/roles/setup-kafka/defaults/main.yml:37]: Do we need 7 days for every topic? That will burn out the disks.

File: ansible/roles/setup-kafka/tasks/main.yml (hunk 3 above)
Comment (keshavprasadms) [reply ansible/roles/setup-kafka/tasks/main.yml:16]: Ok I get it, we have another conditional `ml_kafka_topic_create == true`.

File: ansible/roles/setup-kafka/tasks/main.yml [hunk 5]
```diff
@@ -12,3 +12,11 @@
...
+  command: /opt/kafka/bin/kafka-topics.sh --zookeeper localhost:2181 --create --topic {{env_name}}.{{item.name}} --partitions {{ item.num_of_partitions }} --replication-factor {{ item.replication_factor }}
+  with_items: "{{ml_service_topics}}"
+  ignore_errors: true
+  when: when: kafka_id=="1" and ml_kafka_topic_create == true
```
Comment (keshavprasadms) [inline ansible/roles/setup-kafka/tasks/main.yml:20]: this should be a boolen check `ml_kafka_topic_create is true`
`==` should only be used for strings

**Review by keshavprasadms — CHANGES_REQUESTED** [review]
Please check my comments

File: ansible/roles/setup-kafka/tasks/main.yml (hunk 5 above)
Comment (sivakarthikeyanoptit) [reply ansible/roles/setup-kafka/tasks/main.yml]: changed this variable when: kafka_id=="1" and ml_kafka_topic_create is true

**Review by keshavprasadms — APPROVED** [review]

**Pull request merged**